### DISCUSSION TITLE
Support for conditional compilation (#IFDEF kind of thing)

### DISCUSSION THREAD

**Discussion by OrionGonzalez** [swc-project/swc#5265] [Ideas]
I have been looking around but I couldn't find anything. Is there any support/plans to support C style preprocessor directives?

It would be really useful for me because I need to compile for a lot of different platforms (node, browser, webOS, es5...) and also for debugging purposes. eg. `#IFNDEF __DEBUG console.log = ()=>{} `.

C-style macros and c++ constexpr would be cool too maybe

**Comment by kwonoj** [comment]
No in core transform. These are typically in the territory of custom transform plugin, or bundler plugins.

**Reply by OrionGonzalez** [reply]
Do you know of any plugins that do this?

**Reply by 0xedb** [reply]
@OrionGonzalez any ideas? thanks
